hypershift-operator/controllers/hostedcluster/internal/platform/azure/azure.go

```diff
@@ -184,14 +200,62 @@ func (a *Azure) ReconcileCredentials(ctx context.Context, c client.Client, creat
 	return nil
 }
 
-func (a *Azure) ReconcileSecretEncryption(ctx context.Context, c client.Client, createOrUpdate upsert.CreateOrUpdateFN, hcluster *hyperv1.HostedCluster, controlPlaneNamespace string) error {
+func (a Azure) ReconcileSecretEncryption(ctx context.Context, c client.Client, createOrUpdate upsert.CreateOrUpdateFN, hcluster *hyperv1.HostedCluster, controlPlaneNamespace string) error {
+	return nil
+}
+
+func (a Azure) CAPIProviderPolicyRules() []rbacv1.PolicyRule {
+	return nil
+}
+
+func (a Azure) DeleteCredentials(ctx context.Context, c client.Client, hcluster *hyperv1.HostedCluster, controlPlaneNamespace string) error {
 	return nil
 }
 
-func (a *Azure) CAPIProviderPolicyRules() []rbacv1.PolicyRule {
+func reconcileAzureCluster(azureCluster *capiazure.AzureCluster, hcluster *hyperv1.HostedCluster, apiEndpoint hyperv1.APIEndpoint, controlPlaneNamespace string) error {
+	if azureCluster.Annotations == nil {
+		azureCluster.Annotations = map[string]string{}
+	}
+
+	azureCluster.Annotations[capiv1.ManagedByAnnotation] = "external"
+
+	azureCluster.Spec.Location = hcluster.Spec.Platform.Azure.Location
+	azureCluster.Spec.ResourceGroup = hcluster.Spec.Platform.Azure.ResourceGroupName
+	azureCluster.Spec.NetworkSpec.Vnet.ID = hcluster.Spec.Platform.Azure.VnetID
+	azureCluster.Spec.NetworkSpec.Vnet.Name = hcluster.Spec.Platform.Azure.VnetName
+	azureCluster.Spec.NetworkSpec.Vnet.ResourceGroup = hcluster.Spec.Platform.Azure.ResourceGroupName
+	azureCluster.Spec.SubscriptionID = hcluster.Spec.Platform.Azure.SubscriptionID
+
+	azureCluster.Spec.ControlPlaneEndpoint = capiv1.APIEndpoint{
+		Host: apiEndpoint.Host,
+		Port: apiEndpoint.Port,
+	}
+
+	azureCluster.Status.Ready = true
+
+	azureCluster.Spec.IdentityRef = &corev1.ObjectReference{Name: hcluster.Name, Namespace: controlPlaneNamespace}
+
 	return nil
 }
 
-func (a *Azure) DeleteCredentials(ctx context.Context, c client.Client, hcluster *hyperv1.HostedCluster, controlPlaneNamespace string) error {
+func reconcileAzureClusterIdentity(ctx context.Context, c client.Client, hcluster *hyperv1.HostedCluster, azureClusterIdentity *capiazure.AzureClusterIdentity, controlPlaneNamespace string) error {
+	var userCloudCredSecret corev1.Secret
+
+	credentialsSecret := &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: hcluster.Spec.Platform.Azure.Credentials.Name, Namespace: controlPlaneNamespace}}
+	if err := c.Get(ctx, client.ObjectKeyFromObject(credentialsSecret), &userCloudCredSecret); err != nil {
```

Comment:
```suggestion
	if err := c.Get(ctx, client.ObjectKeyFromObject(credentialsSecret), &credentialsSecret); err != nil {
```
`userCloudCredSecret` var is not needed anymore.